### PR TITLE
Improve documentation about testing modes

### PR DESCRIPTION
**Pull request by Evangelink**
## Summary

Explains how `dotnet test` behaves depending on the mode it is being run with and explicitly mentions extra arguments.

Relates to https://github.com/dotnet/sdk/issues/23198

@nohwnd for review on the technical accuracy.

### CODE REVIEW

File: docs/core/tools/dotnet-test.md [hunk 1]
```diff
@@ -234,6 +251,12 @@ Where `Microsoft.NET.Test.Sdk` is the test host, `xunit` is the test framework.
   dotnet test --blame
   ```
 
+- Run the tests in the `test1` project providing the `-bl` argument down to `msbuild`:
+
+  ```dotnetcli
+  dotnet test ~/projects/test1/test1.csproj -bl
+  ```
```
Comment (nohwnd) [inline docs/core/tools/dotnet-test.md:258]: Also consider adding an example of passing an msbuild property

```suggestion
- Run the tests in the `test1` project providing the `-bl` (binary log) argument down to `msbuild`:

  ```dotnetcli
  dotnet test ~/projects/test1/test1.csproj -bl
  ```
```

Comment (Evangelink) [reply docs/core/tools/dotnet-test.md:258]: Isn't this a msbuild property?

File: docs/core/tools/dotnet-test.md [hunk 2]
```diff
@@ -234,6 +251,11 @@ Where `Microsoft.NET.Test.Sdk` is the test host, `xunit` is the test framework.
   dotnet test --blame
   ```
 
+- Run the tests in the `test1` project providing the `-bl` (binary log) argument down to `msbuild`:
+
+  ```dotnetcli
+  dotnet test ~/projects/test1/test1.csproj -bl
```
Comment (Evangelink) [inline docs/core/tools/dotnet-test.md:257]:
```suggestion
  dotnet test ~/projects/test1/test1.csproj -bl
  ```
```

File: docs/core/tools/dotnet-test.md [hunk 3]
```diff
@@ -234,6 +251,11 @@ Where `Microsoft.NET.Test.Sdk` is the test host, `xunit` is the test framework.
...
+
```
Comment (Evangelink) [inline docs/core/tools/dotnet-test.md:258]:
```suggestion
  ```dotnetcli
  dotnet test ~/projects/test1/test1.csproj -bl
  ```

```

File: docs/core/tools/dotnet-test.md (hunk 1 above)
Comment (KalleOlaviNiemitalo) [reply docs/core/tools/dotnet-test.md]: It's an MSBuild option but not a property.

File: docs/core/tools/dotnet-test.md [hunk 4]
```diff
@@ -234,6 +251,18 @@ Where `Microsoft.NET.Test.Sdk` is the test host, `xunit` is the test framework.
   dotnet test --blame
   ```
 
+- Run the tests in the `test1` project providing the `-bl` (binary log) argument down to `msbuild`:
```
Comment (tdykstra) [inline docs/core/tools/dotnet-test.md:254]:
```suggestion
- Run the tests in the `test1` project, providing the `-bl` (binary log) argument to `msbuild`:
```

File: docs/core/tools/dotnet-test.md [hunk 5]
```diff
@@ -234,6 +251,18 @@ Where `Microsoft.NET.Test.Sdk` is the test host, `xunit` is the test framework.
...
+
+  ```dotnetcli
+  dotnet test ~/projects/test1/test1.csproj -bl  
+  ```
+
+- Run the tests in the `test1` project providing the `-p:DefineConstants="DEV"` argument property down to `msbuild`:
```
Comment (tdykstra) [inline docs/core/tools/dotnet-test.md:260]:
```suggestion
- Run the tests in the `test1` project, setting the MSBuild `DefineConstants` property to `DEV`:
```

**Review by tdykstra — COMMENTED** [review]
Thanks for providing this, I just made a few suggestions to change the wording of the added text.

**Review by tdykstra — APPROVED** [review]

**Pull request merged**